xbmc/addons/gui/GUIDialogAddonSettings.cpp

```diff
@@ -128,7 +132,134 @@ bool CGUIDialogAddonSettings::ShowForAddon(const ADDON::AddonPtr& addon,
   if (!g_passwordManager.CheckMenuLock(WINDOW_ADDON_BROWSER))
     return false;
 
-  if (!addon->HasSettings())
+  bool byNew = false;
+  uint32_t instanceNumber = KODI::ADDONS::ADDON_SETTINGS_ID;
+
+  /*
+   * Below becomes the optional possibility handled within an selection dialog to
+   * select about one add-on given different instances, e.g. by PVR addons to
+   * allow various connections to backends.
+   */
+  if (addon->SupportInstanceSettings())
+  {
+    constexpr int BASE_SETTINGS = 0;
+    constexpr int ADD_INSTANCE = 100;
+    constexpr int DELETE_INSTANCE = 101;
+    constexpr int GENERAL_BUTTON_START = ADD_INSTANCE;
+
+    CGUIDialogSelect* dialog =
+        CServiceBroker::GetGUI()->GetWindowManager().GetWindow<CGUIDialogSelect>(
+            WINDOW_DIALOG_SELECT);
+    if (dialog)
+    {
+      while (true)
+      {
+        CFileItemList itemsGeneral;
+        CFileItemList itemsInstances;
+
+        dialog->Reset();
+        dialog->SetHeading(10012); // Label: "Add-on configuration setting selection"
+        dialog->SetUseDetails(false);
+
+        if (addon->HasSettings(KODI::ADDONS::ADDON_SETTINGS_ID))
+        {
+          // Label: "Base settings"
+          const CFileItemPtr item = std::make_shared<CFileItem>(g_localizeStrings.Get(10013));
+          item->SetProperty("id", BASE_SETTINGS);
+          itemsGeneral.Add(item);
+        }
+        {
+          // Label: "Add add-on configuration"
```

Comment:
i think we changed that string, change the comment to match the new string

Reply:
This is equal, only the other above was gone down and becomes changed.